### PR TITLE
[WIP] *: Fully concurrent large reads

### PR DESCRIPTION
**Pull request by jpbetz**
This is a WIP continuation of https://github.com/etcd-io/etcd/pull/9384.

### CODE REVIEW

File: mvcc/backend/concurrent_read_scheduler.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	// managing context deadlines and cancelations also.
+
+	concurrentReadTxs := r.openReadCounter.Value()
+	for i := uint64(0); i < (r.maxConcurrentReadTxs - concurrentReadTxs); i++ {
+		select {
+		case rch := <-r.readTxCh:
+			rtx, err := r.b.db.Begin(false)
+			if err != nil {
+				plog.Fatalf("cannot begin read tx (%s)", err)
+			}
```
Comment (hexfusion) [inline mvcc/backend/concurrent_read_scheduler.go]: Do we want to also add zap here capnslog will be dropped in 3.4

File: mvcc/backend/concurrent_read_tx.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	tx *bolt.Tx
+}
+
+func (rt *concurrentReadTx) Lock()   {}
+func (rt *concurrentReadTx) Unlock() { rt.tx.Rollback() }
+
+func (rt *concurrentReadTx) UnsafeRange(bucketName, key, endKey []byte, limit int64) ([][]byte, [][]byte) {
+	bucket := rt.tx.Bucket(bucketName)
+	if bucket == nil {
+		plog.Fatalf("bucket %s does not exist", bucketName)
```
Comment (hexfusion) [inline mvcc/backend/concurrent_read_tx.go]: same

File: mvcc/backend/concurrent_read_scheduler.go (hunk 1 above)
Comment (jpbetz) [reply mvcc/backend/concurrent_read_scheduler.go]: Good catch, I'll add it.

**Comment by jpbetz** [comment]
See https://github.com/etcd-io/etcd/pull/10523 for current efforts

**Pull request closed without merging**